pkg/services/ngalert/store/alert_rule.go

```diff
@@ -455,9 +472,28 @@ func (st DBstore) GetAlertRulesForScheduling(ctx context.Context, query *ngmodel
 			alertRulesSql += " WHERE " + filter
 		}
 
-		if err := sess.SQL(alertRulesSql, args...).Find(&rules); err != nil {
+		var faultyRules int64 = 0
+		rule := new(ngmodels.AlertRule)
+		rows, err := sess.SQL(alertRulesSql, args...).Rows(rule)
+		if err != nil {
 			return fmt.Errorf("failed to fetch alert rules: %w", err)
 		}
+		defer rows.Close()
+
+		for rows.Next() {
+			rule := new(ngmodels.AlertRule)
+			err = rows.Scan(rule)
+			if err != nil {
+				faultyRules++
+				continue
+			}
```

Comment:
A faulty rule is a really exceptional situation, let's emit a separate log for each faulty rule. This lets us capture more debug info like the actual error, this can help when operating Grafana
```suggestion
			if err != nil {
				st.Logger.Error("Faulty rule found in DB store, ignoring it", "func", "GetAlertRulesForScheduling", "error", err)
				continue
			}
```